prow/plank/controller.go

```diff
@@ -230,99 +210,140 @@ func (c *Controller) terminateDupes(pjs []kube.ProwJob) error {
 	return nil
 }
 
-func (c *Controller) syncKubernetesJob(pj kube.ProwJob, pm map[string]kube.Pod, reports chan<- kube.ProwJob) error {
-	if pj.Complete() {
-		if pj.Status.PodName == "" {
-			// Completed ProwJob, already cleaned up the pod. Nothing to do.
+// TODO: Dry this out
```

Comment:
Can you say a little more here, since it's not clear to me what needs to be dried out.

Reply:
We have the same helper inside the jenkins-operator, the only difference is that here we are passing a pod map, in jenkins-operator we will be passing a jenkinsbuild map (https://github.com/kubernetes/test-infra/issues/4366). In the grand scheme of things (in order to reduce future refactorings and friction for people who want to support new agents) I am thinking of creating a controller interface and some machinery that will automatically generate a controller given an interface implementation.